rackcity/views/it_model_views.py

```diff
@@ -18,22 +18,31 @@ def model_list(request):
         return JsonResponse(serializer.data, safe=False)
 
 
-def model_add(request):  # TODO: make it only accept POSTs
+@api_view(['POST'])
+@permission_classes([IsAdminUser])
+def model_add(request):
     """
     Add a new model
     """
-    if request.method == 'POST':
-        data = JSONParser().parse(request)
-        serializer = ITModelSerializer(data=data)
-        if serializer.is_valid():  # TODO: figure out what .valid() does.  Different fields are required for add, update, etc
-            print("valid")
+    data = JSONParser().parse(request)
+    failure_message = ""
+    if 'id' in data:
+        failure_message = failure_message + "Don't include id when adding a model. "
+    serializer = ITModelSerializer(data=data)
+    if not serializer.is_valid(raise_exception=False):
+        failure_message = failure_message + str(serializer.errors)
+    if failure_message == "":
+        try:
             serializer.save()
-            # should return something better maybe
-            return HttpResponse(status=201)
-        else:
-            print("invalid")
-            # should return something better definitely
-            return HttpResponse(status=400)
+            return JsonResponse({"success": True})
+        except Exception as error:
+            failure_message = failure_message + str(error)
+
+    failure_message = "Request was invalid. " + failure_message
+    return JsonResponse({
+        "success": False,
+        "failure_message": failure_message
+    })
```

Comment:
**_per our discussion_** I think it would be worthwhile to populate the `status` param on the `JsonResponse` with the corresponding HTTP status code, so we return something like: 

```
JsonResponse(
    {"error_description": "blah"},
    status=400,
)
``` 
on bad requests and 
```
 JsonResponse(
    {},
    status=200,
)
```
on good requests?

Reply:
I think we should use these https://docs.python.org/3/library/http.html#http.HTTPStatus for better readability

Reply:
Sick I'm down